### ISSUE TITLE
pub interoperability with bower

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#1007]
<a href="https://github.com/jmesserly"><img src="https://avatars.githubusercontent.com/u/1081711?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [jmesserly](https://github.com/jmesserly)**
_Originally opened as dart-lang/sdk#19354_

----

http://bower.io/ is becoming popular for client-side libraries. In particular it may become the primary delivery mechanism for custom elements such as https://github.com/Polymer/core-toolbar and https://github.com/x-tag/sparkline

Several constraints that jump out. These are needed for bare-minimum interop:
\* URIs imported with &lt;link rel=import href=...&gt; must be at a canonical location, so they can be imported only once and that URI is shared by all Pub and Bower packages.
\* Bower package names must support dash in the name, for example &quot;x-tag&quot;. Dash is considered idiomatic.

Naturally this is a bit of a long-term goal, but might be worth tracking as it seems important

**Comment by DartBot** [comment]
<a href="https://github.com/jmesserly"><img src="https://avatars.githubusercontent.com/u/1081711?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jmesserly](https://github.com/jmesserly)**

----

*Removed [Priority-Unassigned](../labels/Priority-Unassigned) label.*
*Added [Priority-High](../labels/Priority-High) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

If this is high, it should have an owner and a milestone. Assigning both. :)
______
*Set owner to @justinfagnani.*
*Added this to the [1.6](../milestones/1.6) milestone.*

**Comment by DartBot** [comment]
<a href="https://github.com/jmesserly"><img src="https://avatars.githubusercontent.com/u/1081711?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jmesserly](https://github.com/jmesserly)**

----

Bob & Nathan -- I wasn't sure which of you to assign this to, but as the Pub folks you deserve first crack at it :-)

______
 cc @justinfagnani.
 cc @nex3.
*Set owner to @munificent.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

> as the Pub folks you deserve first crack at it :-)

Yup! This is on our list for this quarter.

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

See also https://code.google.com/p/dart/issues/detail?id=20284 which asks for the Google Web Components to work with Dart. Maybe when we have bower + pub integration, we'll get those elements "for free" ?

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

I've generalized this issue as: https://code.google.com/p/dart/issues/detail?id=20323
______
*Added [NotPlanned](../labels/NotPlanned) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/justinfagnani"><img src="https://avatars.githubusercontent.com/u/522948?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [justinfagnani](https://github.com/justinfagnani)**

----

I don't think the new issue generalizes this, as many libraries aren't web components, and we still need a way for a pub package to depend on a non pub package. I have some proposals or to Bob and Natalie for hooks that would allow this. If those hooks are built, we can implement bower, npm, etc dependencies separate from pub.
